draft-ietf-quic-http.md

```diff
@@ -583,12 +673,34 @@ server to client, as in HTTP/2.  It defines no flags.
 
 The payload consists of:
 
-  Promised Stream ID:
-  : A 32-bit Stream ID indicating the QUIC stream on which the response will be
-    sent
+Push ID:
+: A 32-bit identifier for the server push request.  A push ID is used in push
+  stream header ({{server-push}}), CANCEL_PUSH frames ({{frame-cancel-push}}),
+  and PRIORITY frames ({{frame-priority}}).
+
+Header Block:
+: HPACK-compressed request headers for the promised response.
 
-  Payload:
-  : HPACK-compressed request headers for the promised response.
+A server MAY use the same Push ID in multiple PUSH_PROMISE frames.  This allows
+the server to use the same server push in response to multiple concurrent
+requests.  Referencing the same server push ensures that a PUSH_PROMISE can be
+made in relation to every response in which server push might be needed without
+causing duplicating pushes.
+
+A server that uses the same Push ID in multiple PUSH_PROMISE frames MUST include
+the same header fields each time.  The octets of the header block MAY be
+different due to differing encoding, but the header fields and their values MUST
+be identical.  Note that ordering of header fields is significant.  A client
+MUST treat receipt of a PUSH_PROMISE with conflicting header field values for
+the same Push ID as a connection error of type HTTP_MALFORMED_PUSH_PROMISE.
+
+A server SHOULD avoid sending a PUSH_PROMISE that includes a Push ID that was
+fulfilled prior to the request being made.  Though a client needs to handle
+receiving a promised response prior to it being promised due to reordering of
+stream delivery, clients might be unable to reuse a pushed response that was
+long since consumed. \[Editor's Note: should we include a count of the number of
```

Comment:
A server might promise with Push ID X and fulfill that promise.  Later, it could send the same PUSH_PROMISE for X and assume that the client already has the response.  A client can't be expected to retain the response.  I have a different proposal for handling this now that I've had time away from the problem, I'll write that down.